HeterogeneousCore/AlpakaInterface/interface/geomFunctions.h

```diff
@@ -0,0 +1,57 @@
+#ifndef HeterogeneousCore_AlpakaInterface_interface_geomFunctions_h
+#define HeterogeneousCore_AlpakaInterface_interface_geomFunctions_h
+
+namespace cms::alpakatools {
+
+  template <typename TAcc>
+  ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE float eta(TAcc const& acc, const float x, const float y, const float z) {
+    float r3 = alpaka::math::sqrt(acc, x * x + y * y + z * z);
+    float rt = alpaka::math::sqrt(acc, x * x + y * y);
+    float eta = ((z > 0) - (z < 0)) * alpaka::math::acosh(acc, r3 / rt);
```

Comment:
I don't think this and a few other implementations are acceptable in a common space

Reply:
I meant here mainly the numerical implementation
numerically `0.5f*log((p+pz)/(p-pz))` is more economical.

Reply:
I thought that the relevant https://github.com/cms-sw/cmssw/pull/45117#discussion_r1674782508 was general. Let me know what can be considered general or not.

Reply:
I'd rather decouple updates in the central space from the LST features.
The steps would be : introduce the generic functions first, have them reviewed and approved and then switch LST to use them.
There is a very broad phase space especially with somewhat random math methods in geomFunctions.

Something present in c++ standard and not in alpaka should be more acceptable/straighforward.

Reply:
Sure. Then I revert this and add it to the list of "things to do", mentioning the strategy?